### DISCUSSION TITLE
'Bindable' is unavailable in macOS

### DISCUSSION THREAD

**Discussion by rehatkathuria** [pointfreeco/swift-perception#43] [General]
`@Perception.Bindable` is unavailable on macOS. Is there an alternative to use or is this an oversight?

**Comment by mbrandonw** [comment]
Hi @rehatkathuria, this only shows if you are deploying to macOS 14+, and so you can just use `@Bindable` directly.

**Reply by rehatkathuria** [reply]
Would be good if this wasn't an error because we have a use case where the package has support for the following:

```
    platforms: [
        .iOS(.v15),
        .macOS(.v14),
    ],
```

I'm currently using a compiler flag but feel like it makes more sense for the library to handle this for consistency at the call sites?

```
    #if os(macOS)
        @Bindable var model: FeedbackServiceModel
    #else
        @Perception.Bindable var model: FeedbackServiceModel
    #endif
```

**Reply by mbrandonw** [reply]
You can define a type alias in your project to unify them, but we don't want to add it to the library because we'd have to figure out a reasonable name. I think that's a tall order, whereas a simple type alias in your app doesn't have such a high bar, and so really can be anything.

I did just add some better messaging to the unavailability that will hopefully make it clearer how to fix: https://github.com/pointfreeco/swift-perception/pull/45/commits/cb5084aba88bf6f20b16cfd3f93bfe96c75086cb
